lib/elixir/lib/integer.ex

```diff
@@ -400,17 +400,23 @@ defmodule Integer do
       iex> Integer.gcd(10, 0)
       10
 
+      iex> Integer.gcd(7, 7)
+      7
+
       iex> Integer.gcd(0, 0)
       0
 
   """
+  @spec gcd(0, 0) :: 0
+  @spec gcd(int, int) :: int when int: pos_integer
```

Comment:
If I understand correctly, what you're saying is that `fun(a, a) when a: type` is the same as `fun(type, type)`, but the erlang docs say something exactly opposite:

>     -spec id(X) -> X when X :: tuple().
>
>[...]
>The above function specification uses multiple occurrences of the same type variable. That provides more type information than the following function specification, where the type variables are missing:
>
>     -spec id(tuple()) -> tuple().
>
>The latter specification says that the function takes some tuple and returns some tuple. The specification with the X type variable specifies that the function takes a tuple and returns **the same** tuple.